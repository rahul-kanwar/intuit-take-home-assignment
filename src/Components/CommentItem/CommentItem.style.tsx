import styled from "styled-components";

export const Item = styled.div`
  border-bottom: 1px solid #ccc;
  padding: 10px 0;
`;