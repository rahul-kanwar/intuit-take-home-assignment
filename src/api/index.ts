import axios from 'axios';

export const apiClient = axios.create({
  baseURL: 'http://localhost:3001',
});

export interface Comment {
  id: number;
  name: string;
  created: string;
  message: string;
}

export const getComments = async (): Promise<Comment[]> => {
  const response = await apiClient.get('/getComments');
  return response.data;
};

export const createComment = async (comment: Omit<Comment, 'id'>): Promise<Comment> => {
  const response = await apiClient.post('/createComment', comment);
  return response.data;
};

export const deleteComments = async (): Promise<void> => {
  await apiClient.delete('/deleteComments');
};
