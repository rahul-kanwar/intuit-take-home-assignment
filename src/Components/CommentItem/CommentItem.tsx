import React from 'react';
import { Comment } from '../../api';
import { format } from 'date-fns';
import { Item } from './CommentItem.style';

const CommentItem: React.FC<{ comment: Comment }> = ({ comment }) => {
    let formattedDate = 'Invalid date';
    try {
      if (comment.created) {
        console.log('Parsing date:', comment.created); // Log the date string
        // Parse the date using the provided format
          formattedDate = format(new Date(comment.created), 'Pp');
      }
    } catch (error) {
      console.error('Error parsing date:', error);
    }
    return (
        <Item>
            <h4>{comment.name}</h4>
            <p>{comment.message}</p>
            <small>{formattedDate}</small>
        </Item>
    );
};

export default CommentItem;
