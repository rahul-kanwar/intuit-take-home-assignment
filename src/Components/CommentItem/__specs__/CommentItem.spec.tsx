import React from 'react';
import { render, screen } from '@testing-library/react';

import 'jest-styled-components';
import { Comment } from '../../../api';
import CommentItem from '../CommentItem';

// Mock comment data
const mockComment: Comment = {
  id: 1,
  name: 'John Doe',
  message: 'This is a test comment',
  created: '2024-06-26 10:35:42',
};

const mockInvalidDateComment: Comment = {
  id: 2,
  name: 'Jane Doe',
  message: 'This comment has an invalid date',
  created: 'invalid date string',
};

describe('CommentItem Component', () => {
  it('renders correctly with valid date', () => {
    const { asFragment } = render(<CommentItem comment={mockComment} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with invalid date', () => {
    const { asFragment } = render(
      <CommentItem comment={mockInvalidDateComment} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('displays the correct content with valid date', () => {
    render(<CommentItem comment={mockComment} />);
    expect(screen.getByText('John Doe')).toBeInTheDocument();
    expect(screen.getByText('This is a test comment')).toBeInTheDocument();
    expect(screen.getByText(/6\/26\/2024/)).toBeInTheDocument();
  });

  it('displays "Invalid date" with invalid date', () => {
    render(<CommentItem comment={mockInvalidDateComment} />);
    expect(screen.getByText('Jane Doe')).toBeInTheDocument();
    expect(
      screen.getByText('This comment has an invalid date')
    ).toBeInTheDocument();
    expect(screen.getByText('Invalid date')).toBeInTheDocument();
  });
});
