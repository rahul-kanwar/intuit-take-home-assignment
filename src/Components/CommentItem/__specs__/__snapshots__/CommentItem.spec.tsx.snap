// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentItem Component renders correctly with invalid date 1`] = `
<DocumentFragment>
  .c0 {
  border-bottom: 1px solid #ccc;
  padding: 10px 0;
}

<div
    class="c0"
  >
    <h4>
      Jane Doe
    </h4>
    <p>
      This comment has an invalid date
    </p>
    <small>
      Invalid date
    </small>
  </div>
</DocumentFragment>
`;

exports[`CommentItem Component renders correctly with valid date 1`] = `
<DocumentFragment>
  .c0 {
  border-bottom: 1px solid #ccc;
  padding: 10px 0;
}

<div
    class="c0"
  >
    <h4>
      John Doe
    </h4>
    <p>
      This is a test comment
    </p>
    <small>
      06/26/2024, 10:35 AM
    </small>
  </div>
</DocumentFragment>
`;
